### PR TITLE
Resolve all swiftlint violations.

### PR DESCRIPTION
**Pull request by jakeadams**
Some clean up of the compilation warnings. Most of this is either adding `swiftlint:disable` hints or resolving the issue.

In collaboration with @mdiep, I disabled 2 rules in `.swiftlint.yml` definition.
I also removed conformance to `CustomStringConvertible` for `Platform` as it only returned the `rawValue` and the call site was accessing `description` directly, which defeats the purpose of the protocol. I updated the call site to use `rawValue` directly. I added a test around this to make sure nothing changed.

### CODE REVIEW

File: Tests/CarthageKitTests/XcodeSpec.swift [hunk 1]
```diff
@@ -316,6 +316,7 @@ class XcodeSpec: QuickSpec {
 			}
 
 			expect(isExpectedError) == true
+			expect(result.error?.description) == "Dependency \"Swell-0.5.0\" has no shared framework schemes for any of the platforms: Mac"
```
Comment (jdhealy) [inline Tests/CarthageKitTests/XcodeSpec.swift]: Was this change intended for this PR?

File: Source/CarthageKit/Project.swift [hunk 2]
```diff
@@ -919,10 +917,10 @@ public final class Project { // swiftlint:disable:this type_body_length
 		let fileManager = FileManager.default
 
 		return self.dependencySet(for: dependency, version: version)
-			.zip(with: // file system objects which might conflict with symlinks
-				list(treeish: version.commitish, atPath: carthageProjectCheckoutsPath, inRepository: repositoryURL)
-					.map { (path: String) in (path as NSString).lastPathComponent }
-					.collect()
+			// file system objects which might conflict with symlinks
+			.zip(with: list(treeish: version.commitish, atPath: carthageProjectCheckoutsPath, inRepository: repositoryURL)
```
Comment (jdhealy) [inline Source/CarthageKit/Project.swift]: The linting message I see (*“Colon Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)”*) is mistaken in that this is a method call with an argument label.

We should just use `// swiftlint:disable:previous colon` on the next line to workaround this errant message from the linter.

**Comment by jdhealy** [comment]
Great changes overall 👏

File: Tests/CarthageKitTests/XcodeSpec.swift (hunk 1 above)
Comment (blender) [reply Tests/CarthageKitTests/XcodeSpec.swift]: Yes. This is indented to test the `description` -> `rawValue` change to Platform

Comment (jdhealy) [reply Tests/CarthageKitTests/XcodeSpec.swift]: Ah, okay. Well, divorced from the potential of these `XcodeSpec` changes being removed, I'm fine enough with the changes to [the `zip(with: //` thing](https://github.com/Carthage/Carthage/pull/2374#discussion_r174268071).

**Comment by jdhealy** [comment]
bravo

**Comment by jakeadams** [comment]
Thank you, all!
